cpp/betweenness_centrality_module/algorithm_online/betweenness_centrality_online.cpp

```diff
@@ -0,0 +1,530 @@
+#include <mg_graph.hpp>
+
+#include "../../biconnected_components_module/algorithm/biconnected_components.hpp"
+#include "../algorithm/betweenness_centrality.hpp"
+#include "betweenness_centrality_online.hpp"
+
+template <typename T>
+void RemoveDuplicates(std::vector<T> &vector) {
+  std::unordered_set<T> seen;
+
+  auto new_end = std::remove_if(vector.begin(), vector.end(), [&seen](const T &value) {
+    if (seen.count(value)) return true;
+
+    seen.insert(value);
+    return false;
+  });
+
+  vector.erase(new_end, vector.end());
+}
+
+namespace online_bc {
+bool OnlineBC::Inconsistent(const mg_graph::GraphView<> &graph) const {
+  if (graph.Nodes().size() != this->node_bc_scores.size()) return true;
+
+  for (const auto [node_inner_id] : graph.Nodes()) {
+    if (!this->node_bc_scores.count(graph.GetMemgraphNodeId(node_inner_id))) return true;
+  }
+
+  return false;
+}
+
+std::unordered_map<std::uint64_t, double> OnlineBC::NormalizeBC(
+    const std::unordered_map<std::uint64_t, double> &node_bc_scores, const std::uint64_t graph_order) const {
+  const double normalization_factor = 2.0 / ((graph_order - 1) * (graph_order - 2));
+  std::unordered_map<std::uint64_t, double> normalized_bc_scores;
+  for (const auto [node_id, bc_score] : node_bc_scores) {
+    normalized_bc_scores[node_id] = bc_score * normalization_factor;
+  }
+
+  return normalized_bc_scores;
+}
+
+void OnlineBC::CallBrandesAlgorithm(const mg_graph::GraphView<> &graph, const std::uint64_t threads) {
+  this->node_bc_scores.clear();
+
+  const auto bc_scores = betweenness_centrality_alg::BetweennessCentrality(graph, false, false, threads);
+  for (std::uint64_t node_id = 0; node_id < graph.Nodes().size(); ++node_id) {
+    this->node_bc_scores[graph.GetMemgraphNodeId(node_id)] = bc_scores[node_id];
+  }
+}
+
+std::tuple<std::unordered_map<std::uint64_t, std::uint64_t>, std::unordered_map<std::uint64_t, std::set<std::uint64_t>>,
+           std::vector<std::uint64_t>>
```

Comment:
Valid :+1:

Reply:
What I had in mind is something like 
```
struct BFSResult {
std::tuple<std::unordered_map<std::uint64_t, std::uint64_t> n_shortest_paths;
std::unordered_map<std::uint64_t, std::uint64_t> distances;
std::unordered_map<std::uint64_t, std::set<std::uint64_t>> predecessors;
std::vector<std::uint64_t>> bfs_order;
}
```
It would be much clearer because results would be standardized. Do this If you think it is feasible here, if it adds too much bad code, skip this comment.

Reply:
Done! The modified code is now much clearer. :slightly_smiling_face: